src/main/java/sirius/kernel/xml/StructuredNode.java

```diff
@@ -143,6 +149,65 @@ public Node getNode() {
         return resultList;
     }
 
+    /**
+     * Returns a map of all attributes of this DOM node with their names as keys.
+     *
+     * @return a list containing all attributes of this node. If no attributes exist, an empty list will be returned.
+     */
+    @Nonnull
+    public ImmutableMap<String, StructuredNode> getAttributes() {
```

Comment:
wouldn't it be the best practice to use `Map<...>` (the interface) instead of `ImmutableMap<...>` (a more specific implementation)?

Reply:
ok, for ImmutableMaps this seems to be a more complicated issue...
https://stackoverflow.com/questions/38087900/is-it-better-to-return-an-immutablemap-or-a-map 

you can leave it like that, in my opinion

Reply:
Collections.unmodifiableMap()?